tests/babel-loader.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * NOTE: This ESM module is being used as a node loader (https://nodejs.org/api/esm.html#esm_experimental_loaders)
+ * while running mocha tests, e.g.:
+ *
+ *     npx mocha -n "loader=./tests/babel-loader.js" -r tests/setup.js tests/unit/test.config.js
+ *
+ * it is responsable to transpile on the fly the tests module using babel (otherwise the flowtype syntaxes
+ * will trigger syntax errors while loaded by node plain ESM module loader).
+ *
+ * The simplified transformSource function that follows has been derived from the existing node ESM loader:
+ *
+ * - https://github.com/giltayar/babel-register-esm
+ *
+ * COMPATIBILITY NOTES:
+ *
+ * nodejs module loader API is experimental and so different nodejs versions expects
+ * a different set of hooks:
+ *
+ * - in nodejs < 16: `getSource` and `transformSource`
+ * - in nodejs >= 16: `resolve` and `load`
```

Comment:
😞

Reply:
hehe, I told you that I got "multiple source of fun" in the process of adapting web-ext to type module :rofl: